src/components/NotesPage/NotesPageSagas.js

```diff
@@ -0,0 +1,45 @@
+import { all, call, put, takeEvery } from 'redux-saga/dist/redux-saga-effects-npm-proxy.cjs';
+
+import {
+  CHANGE_USER_HELP_SEEN_ACTION,
+  REQUEST_NOTE_TEXT_ACTION,
+  changeUserHelpSeenAction,
+  changeNoteTextAction,
+  changeSelectedNoteAction
+} from 'components/NotesPage/NotesPageActions';
+import thoughtStorage from 'storage/thought-storage';
+import { APP_INIT_SUCCESS } from 'components/App/AppActions';
+
+const HELP_VIEWED_KEY = 'controls_help:viewed';
+
+function* saveHelpSeenState({ data }) {
+  yield call([localStorage, localStorage.setItem], HELP_VIEWED_KEY, data);
+}
+
+function* readHelpSeenState() {
+  const value = Boolean(
+    JSON.parse(
+      yield call([localStorage, localStorage.getItem], HELP_VIEWED_KEY)
+    )
+  );
+  yield put(changeUserHelpSeenAction(value));
+}
+
+function* setRootNote() {
+  yield put(changeSelectedNoteAction(thoughtStorage.getRoot()));
+}
+
+function* requestNoteText({ data }) {
+  const noteText = yield call(thoughtStorage.getThoughtContent, data);
+  console.debug('ThoughtContentController.loadThought(), loaded thought content: ', noteText);
+  yield put(changeNoteTextAction(noteText));
```

Comment:
Why not "Note text fetch success" action?

In another place we have `UPDATE_NOTE_CONTENT_ACTION`. The name is basically the same as `CHANGE_NOTE_TEXT_ACTION` (changing note text and updating note content is the same).

But these two actions do different things.

Reply:
changeNoteTextAction using in another place to update loading message.
UPDATE_NOTE_CONTENT_ACTION changed to SAVE_NOTE_CONTENT_ACTION